### PR TITLE
I352

### PR DESCRIPTION
**Pull request by littledice**
Demographics spec for test download

### CODE REVIEW

File: spec/DemographicDataset.schema.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"id": "DemographicDataSet",
+	"type": "object",
+    "properties": {
```
Comment (MartinEden) [inline spec/DemographicDataset.schema.json]: Mixed spaces/tabs here?

File: spec/spec.md [hunk 2]
```diff
@@ -1688,3 +1688,76 @@ Schema: [`TouchstoneOverview.schema.json`](TouchstoneOverview.schema.json)
...
+
+#### format
+Optional. A format to return the CSV in, either `wide` or `long`. Defaults to `long`.
+
+### Example
+`/touchstones/2017-op-1/demographics/tot_pop/?format=wide&gender=female`
+
+Total population:
+
+                                       "people"
```
Comment (MartinEden) [inline spec/spec.md]: What is this "people" header? Doesn't that break the CSV format?

Comment (weshinsley) [inline spec/spec.md]: It looks like just a helpful header to say what the numbers mean (besides deriving it from the filename)

CSV is very relaxed - readers just have to skip the right number of header rows. You could prefix with "","","", to encourage it to line up with the first data column.

Source-wise, it's the "name" column of the demographic_value_unit table, where id matches the demographic_value_unit field in the demographic_statistic_type table.

**Comment by MartinEden** [comment]
Actually, the schema check if failing because we haven't merge in #28. Perhaps you could prioritise that, and it would make it clearer if this branch is okay.

File: spec/spec.md (hunk 2 above)
Comment (littledice) [inline spec/spec.md]: Yup, exactly. Lining it up sounds like the right thing to do, will amend.

**Comment by littledice** [comment]
Ok, there were some issues with the schema, fixed now. The demographic type codes in the db have underscores in them, but the `StandardIdentifier` schema only allows hyphens. So we should either
* change codes in db to match schema
* convert underscores to hyphens in the code
* use different schema for demographic stat type codes

We should raise a ticket to do one of these when this PR is merged.

**Pull request merged**